### PR TITLE
eHouse BMS / Home Automation System Integration

### PR DESCRIPTION
**Pull request by ehousepro**
Integrate eHouse BMS, LAN, WiFi, RS-485, Thermostat into Domoticz (according revision suggestions)
- Compiled Win7, 10, Linux x64_x86 (Ubuntu), Raspberry PI 1, 2, 3
- Tested Win7, 10, Linux x64_x86 (Ubuntu),  Raspberry PI  , 2, 3
Functionality:
- Auto Discovery of eHouse  controllers and devices (I/O) after adding eHouse Hardware - via UDP & Update DB
- receive UDP statuses from controllers locally & update DB
- send commands/events to controllers via TCP/IP (sockets)
- receive TCP/IP statuses (optional for remote installation) - including internet

Config: 
- default IP: 192.168.0.200 - eHouse PRO server necessary for NON IP controllers as gateway or connection via Internet 
- default port (TCP) -9876
- default password - abBrac
- default UDP port 6789 (non editable)

- Full description in previous pull request
DIY Blog: http://smart-home.ehouse.pro/category/domoticz/
Controllers info http://eHouse.Biz/

### CODE REVIEW

File: msbuild/domoticz.vcxproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <ProjectGuid>{6AD96441-5B2C-467B-AAB6-A1B8368D503A}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
     <RootNamespace>domoticz</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.15063.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0.16299.0</WindowsTargetPlatformVersion>
```
Comment (gizmocuz) [inline msbuild/domoticz.vcxproj]: Please undo this change

File: msbuild/domoticz.vcxproj [hunk 2]
```diff
@@ -44,9 +44,12 @@
     <LinkIncremental>true</LinkIncremental>
     <IntDir>$(Configuration)_Build\</IntDir>
     <PreBuildEventUseInBuild>true</PreBuildEventUseInBuild>
-    <IncludePath>$(IncludePath)</IncludePath>
+    <IncludePath>$(IncludePath);</IncludePath>
```
Comment (gizmocuz) [inline msbuild/domoticz.vcxproj]: Please undo this change

File: msbuild/domoticz.vcxproj [hunk 3]
```diff
@@ -44,9 +44,12 @@
...
     <CodeAnalysisRuleSet>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Team Tools\Static Analysis Tools\Rule Sets\NativeRecommendedRules.ruleset</CodeAnalysisRuleSet>
     <RunCodeAnalysis>false</RunCodeAnalysis>
+    <LibraryPath>$(VC_LibraryPath_x86);$(WindowsSDK_LibraryPath_x86);$(NETFXKitsDir)Lib\um\x86;</LibraryPath>
```
Comment (gizmocuz) [inline msbuild/domoticz.vcxproj]: Please undo this change

File: msbuild/domoticz.vcxproj [hunk 4]
```diff
@@ -44,9 +44,12 @@
...
+    <SourcePath>$(VC_SourcePath);</SourcePath>
```
Comment (gizmocuz) [inline msbuild/domoticz.vcxproj]: Please undo this change

File: msbuild/domoticz.vcxproj [hunk 5]
```diff
@@ -44,9 +44,12 @@
...
+    <OutDir>d:\eHouseDevelopment\domoticz\msbuild\</OutDir>
```
Comment (gizmocuz) [inline msbuild/domoticz.vcxproj]: Please undo this change

File: www/app/HardwareController.js [hunk 6]
```diff
@@ -362,14 +363,23 @@ define(['app'], function (app) {
 			}
 			else if (
 				(text.indexOf("LAN") >= 0 && ((text.indexOf("YouLess") >= 0) || (text.indexOf("Denkovi") >= 0))) ||
-				(text.indexOf("Relay-Net") >= 0) || (text.indexOf("Satel Integra") >= 0) || (text.indexOf("Harmony") >= 0) || (text.indexOf("Xiaomi Gateway") >= 0) || (text.indexOf("MyHome OpenWebNet with LAN interface") >= 0)
+				(text.indexOf("Relay-Net") >= 0) || (text.indexOf("Satel Integra") >= 0) || (text.indexOf("eHouse") >= 0) || (text.indexOf("Harmony") >= 0) || (text.indexOf("Xiaomi Gateway") >= 0) || (text.indexOf("MyHome OpenWebNet with LAN interface") >= 0)
 			) {
 				var address = $("#hardwarecontent #divremote #tcpaddress").val();
+				if (text.indexOf("eHouse") >= 0)
```
Comment (gizmocuz) [inline www/app/HardwareController.js]: Small request, please place the opening bracket { behind the if as this is done (almost i hope) elsewhere

**Review by gizmocuz — COMMENTED** [review]
Thank you for this ! Some small comments

File: msbuild/domoticz.vcxproj (hunk 1 above)
Comment (ehousepro) [reply msbuild/domoticz.vcxproj]: Can I update once than add it to .gitignore - so it will not be updating any longer?
Without this version and libraries it will not compile and run on VS2017 (I use most recent VS and Windows SDK)

Comment (gizmocuz) [reply msbuild/domoticz.vcxproj]: Unfortunately with gitignore it will ignore the file completely.
But you can modify the file locally, only when you commit your changes, make sure not to commit this file.
As you are using VS2017 (as do i), you can use the 'vstudio tortoisegit toolbar' or use tortuisegit

**Comment by ehousepro** [comment]
TCP IP eHouse AutoDetect and internet connection optimalization

**Comment by gizmocuz** [comment]
Thanks for the great work and addition!

**Comment by ehousepro** [comment]
Hi,

Thank You Very much for merging,

Your SDK version 10.0.15063.0 seems to have a problems, altrough i have it installed -  it cant find files eg: “qos.h” (it is only available qos2.h)

I have to copy 10.0.16299.0 files to 10.0.15063.0 to fullfill dependencies so new SDK works under current domoticz.vcxproj


PS. In a free time I will make some wiki for eHouse system

Regards,


Robert

**Comment by gizmocuz** [comment]
There might be a compilation issue:
https://github.com/domoticz/domoticz/issues/1967
